c_wrapper/lua-mongoc-collection.c

```diff
@@ -650,53 +650,66 @@ lua_mongo_collection_create_index(lua_State *L)
     int keys_index = 3;
     int options_index = 4;
 
-    if (!(lua_istable(L, keys_index))) {
+    if(!(lua_istable(L, keys_index))){
         luaL_error(L, "keys parameter must be a table");
-    }
-
-    if (!(lua_isnil(L, keys_index))){
+    }else{
         throw_error = !(lua_table_to_bson(L, &keys, keys_index, false, absolute_luaBSONObjects_index, &error ));
         if(throw_error){
             goto DONE;
         }
+        lua_pushvalue(L, keys_index);
+        lua_pushnil(L);
+        while(lua_next(L, -2)){
+            lua_pushvalue(L, -2);
+            const char *key = lua_tostring(L, -1);
+            const char *value = lua_tostring(L, -2);
+            if(strlen(idxName) == 0){
+                asprintf(&idxName, "%s_%s", key, value);
+            }else{
+                asprintf(&idxName, "%s_%s_%s", idxName, key, value);
```

Comment:
This is a memory leak. since `asprintf` allocates memory for you, repeated calls to this will change the memory address that the `idxName` points to.

A non memory leak version of this would be:

```
char * temp = idxName;
free(idxName)
asprintf(&idxName, "%s_%s_%s", temp, key, value);
```